autotest/ogr/ogr_mongodbv3.py

```diff
@@ -853,6 +853,32 @@ def test_ogr_mongodbv3_2():
     gdal.PopErrorHandler()
     assert ret != 0
 
+    # Upsert when feature does not exist
```

Comment:
the comment is misleading. The reason of the failure is that the dataset connection is opened in read-only mode at line 652. So the tests from line 862 to 881 don't test anything new.

Tests for nominal situations should be done, with actual modification of a field, at a place where the dataset is opened in update mode. for example towards line 321